### ISSUE TITLE
How to Integration Test State Store

### ISSUE THREAD

**Issue by OnamChilwan** [dapr/dapr#7797]
## Integration Testing State Store

I have a fairly simple application involving a query talking to Cosmos to retrieve some data. The query works fine when I host the application and use something like Postman to hit it over HTTP, however the challenge I am facing is how to write a traditional integration test?

## Using Dapr
``` mermaid
flowchart TD
    A[Query] --> B(Side Car)
    B --> C[Cosmos]
```

The challenge I am facing here is I am running a simple integration test by invoking my query but I get the following error

```
Dapr.DaprException : State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.
  ----> Grpc.Core.RpcException : Status(StatusCode="Unavailable", Detail="Error connecting to subchannel.", DebugException="System.Net.Sockets.SocketException: Connection refused")
  ----> System.Net.Sockets.SocketException : Connection refused
```

The integration test utilises https://github.com/riezebosch/Dapr.Wrapr which is used to start and stop the sidecar. I guess do I need to run the entire application to run this test as the sidecar can only be accessed from the running app? 

To make this feasible, I am assuming I need to run the app using something like `HostBuilder` to run the app and then exercise the query via hitting the app which in this case is over HTTP. For example, it becomes:

```mermaid
flowchart TD
    A[Integration Test]
    A --> B[Dapr Run App]
    B --> C[Seed Data]
    C --> D[Query App Over HTTP]
    D --> E[Assertion]
  
```

This does mean the scope of the integration test is larger than just between the Query and the target.

## The Traditional Approach

``` mermaid
flowchart TD
    A[Query] --> B(Cosmos)
```

Traditionally, this is very simple to achieve as all I need is a connection string to hit Cosmos.

## In a Nutshell

I guess due to the nature of how the sidecar is hidden there is no way of running a test by simply spinning it up and hitting it?
